### PR TITLE
API Make FormField::hasClass return a boolean instead of an int

### PR DESCRIPTION
**Pull request by robbieaverill**
It's always bothered me that this method returns an integer. If nobody has a concern with this change being made, this PR makes it return a boolean.

### CODE REVIEW

File: src/Forms/FormField.php [hunk 1]
```diff
@@ -1247,17 +1247,19 @@ public function transform(FormTransformation $transformation)
...
+     * Returns whether the current field has the given class added
+     *
      * @param string $class
      *
-     * @return int
+     * @return bool
      */
     public function hasClass($class)
     {
         $patten = '/' . strtolower($class) . '/i';
```
Comment (tractorcow) [inline src/Forms/FormField.php]: I'm surprised a method this bad still exists; Not that I'm blaming you in any way @robbieaverill :).

I notice it assumes `magic-bananas` contains the class `banana`. =/ Easier to split on whitespace and just use in_array, wouldn't it?

The `static::class` exception is also super wrong and out of date.

Comment (robbieaverill) [inline src/Forms/FormField.php]: Good time to update! Coming shortly...

**Comment by tractorcow** [comment]
Can be merged on green by anyone with privileges. :P

File: tests/php/Forms/FormFieldTest.php [hunk 2]
```diff
@@ -373,4 +373,15 @@ public function testGetSchemaStateWithFormValidation()
...
     }
+
+    public function testHasClass()
+    {
+        $field = new FormField('Test');
+        $field->addExtraClass('foo BAr cool-banana');
+
+        $this->assertTrue($field->hasClass('foo'));
+        $this->assertTrue($field->hasClass('bAr'));
+        $this->assertFalse($field->hasClass('banana'));
```
Comment (tractorcow) [inline tests/php/Forms/FormFieldTest.php]: Yes, we have no banana.

**Pull request merged**